tests/googletest.build.cmd

```diff
@@ -37,7 +36,7 @@ if not defined CMD_CL (
 )
 set CMD_CL=%CMD_CL:\=/%
 
-cmake %GENERATOR% -DCMAKE_BUILD_TYPE=%CONFIGURATION% ^
+cmake -G "%GENERATOR%" -DCMAKE_BUILD_TYPE=%CONFIGURATION% ^
```

Comment:
`%GENERATOR%` がダブルクォートで囲まれていますが、それを外して `tests1` プロジェクトをビルドすると下記のように出力されました。

<details>
<summary>Output</summary>

```
1>------ Build started: Project: HeaderMake, Configuration: Debug Win32 ------
2>------ Build started: Project: MakefileMake, Configuration: Debug Win32 ------
1>HeaderMake.cpp
2>MakefileMake.cpp
1>HeaderMake.vcxproj -> C:\projects\sakura\Win32\Debug\HeaderMake.exe
2>MakefileMake.vcxproj -> C:\projects\sakura\Win32\Debug\MakefileMake.exe
3>------ Build started: Project: sakura, Configuration: Debug Win32 ------
3>=======================
3>preBuild
3>=======================
3>
3>---- HeaderMake ----
3>Funccode_x.hsrc
3>
3>START HeaderMake.
3>CMDLINE: C:\projects\sakura\Win32\Debug\HeaderMake -in=..\sakura_core\Funccode_x.hsrc -out=..\sakura_core\Funccode_define.h -mode=define
3>
3>
3>END HeaderMake.
3>
3>Funccode_x.hsrc
3>
3>START HeaderMake.
3>CMDLINE: C:\projects\sakura\Win32\Debug\HeaderMake -in=..\sakura_core\Funccode_x.hsrc -out=..\sakura_core\Funccode_enum.h -mode=enum -enum=EFunctionCode
3>
3>
3>END HeaderMake.
3>
3>
3>---- MakefileMake ----
3>
3>START MakefileMake.
3>CMDLINE: C:\projects\sakura\Win32\Debug\MakefileMake -file=..\sakura_core\Makefile -dir=..\sakura_core
3>
3>... Wrote 369 object file lines to tmpfile[..\sakura_core\mfa08816].
3>Makefile needs no change.
3>
3>END MakefileMake.
3>
3>
3>---- Make githash.h ----
3>find-tools.bat
3>|- CMD_GIT=C:\Program Files\Git\cmd\git.exe
3>|- CMD_7Z=C:\Program Files\7-Zip\7z.exe
3>|- CMD_HHC=C:\Program Files (x86)\HTML Help Workshop\hhc.exe
3>|- CMD_ISCC=C:\Program Files (x86)\Inno Setup 5\ISCC.exe
3>|- CMD_CPPCHECK=C:\Program Files\Cppcheck\cppcheck.exe
3>|- CMD_DOXYGEN=
3>|- CMD_VSWHERE=C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe
3>|- CMD_MSBUILD=C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\amd64\MSBuild.exe
3>|- CMD_CMAKE=C:\Program Files\CMake\bin\cmake.exe
3>end
3>checking GIT_SHORT_COMMIT_HASH, GIT_COMMIT_HASH
3>GIT_SHORT_COMMIT_HASH : 345561786
3>GIT_COMMIT_HASH       : 345561786408092d636f517ff9b90f4d2408cc91
3>TEMP_GIT_SHORT_COMMIT_HASH :
3>TEMP_GIT_COMMIT_HASH       :
3>GIT_REMOTE_ORIGIN_URL :
3>GIT_TAG_NAME          :
3>APPVEYOR_URL          :
3>APPVEYOR_REPO_NAME    :
3>APPVEYOR_ACCOUNT_NAME :
3>APPVEYOR_PROJECT_SLUG :
3>APPVEYOR_BUILD_VERSION:
3>APPVEYOR_BUILD_NUMBER :
3>GITHUB_COMMIT_URL           :
3>GITHUB_COMMIT_URL_PR_HEAD   :
3>APPVEYOR_SHORTHASH_PR_HEAD  :
3>APPVEYOR_BUILD_URL          :
3>        1 個のファイルを移動しました。
3>..\sakura_core\githash.h was updated.
3>StdAfx.cpp
3>StdApi.cpp
3>StdControl.cpp
3>CLaxInteger.cpp
3>CMyPoint.cpp
3>CMyRect.cpp
3>CMySize.cpp
3>CMyString.cpp
3>CStrictInteger.cpp
3>CStrictPoint.cpp
3>CStrictRange.cpp
3>CStrictRect.cpp
3>SakuraBasis.cpp
3>CAutoReloadAgent.cpp
3>CAutoSaveAgent.cpp
3>CBackupAgent.cpp
3>CCodeChecker.cpp
3>CDataProfile.cpp
3>CDicMgr.cpp
3>CEditApp.cpp
3>CEol.cpp
3>CFileExt.cpp
3>CGrepAgent.cpp
3>CCesu8.cpp
3>CCodeBase.cpp
3>CCodeFactory.cpp
3>CCodeMediator.cpp
3>CCodePage.cpp
3>CESI.cpp
3>CEuc.cpp
3>charcode.cpp
3>charset.cpp
3>CJis.cpp
3>CLatin1.cpp
3>codechecker.cpp
3>codeutil.cpp
3>CShiftJis.cpp
3>CUnicode.cpp
3>CUnicodeBe.cpp
3>CUtf7.cpp
3>CUtf8.cpp
3>CHokanMgr.cpp
3>CKeyWordSetMgr.cpp
3>CLoadAgent.cpp
3>CMarkMgr.cpp
3>c:\projects\sakura\sakura_core\charset\cesi.cpp(557): warning C5033: 'register' is no longer a supported storage class
3>CViewCommander.cpp
3>CViewCommander_Bookmark.cpp
3>CViewCommander_Clipboard.cpp
3>CViewCommander_Convert.cpp
3>CViewCommander_Cursor.cpp
3>CViewCommander_CustMenu.cpp
3>CViewCommander_Diff.cpp
3>CViewCommander_Edit.cpp
3>CViewCommander_Edit_advanced.cpp
3>CViewCommander_Edit_word_line.cpp
3>CViewCommander_File.cpp
3>CViewCommander_Grep.cpp
3>CViewCommander_Insert.cpp
3>CViewCommander_Macro.cpp
3>CViewCommander_ModeChange.cpp
3>CViewCommander_Outline.cpp
3>CViewCommander_Search.cpp
3>CViewCommander_Select.cpp
3>CViewCommander_Settings.cpp
3>CViewCommander_Support.cpp
3>CViewCommander_TagJump.cpp
3>CViewCommander_Window.cpp
3>build_config.cpp
3>CConvert.cpp
3>CConvert_HaneisuToZeneisu.cpp
3>CConvert_HankataToZenhira.cpp
3>CConvert_HankataToZenkata.cpp
3>CConvert_SpaceToTab.cpp
3>CConvert_TabToSpace.cpp
3>CConvert_ToHankaku.cpp
3>CConvert_ToLower.cpp
3>CConvert_ToUpper.cpp
3>CConvert_ToZenhira.cpp
3>CConvert_ToZenkata.cpp
3>CConvert_Trim.cpp
3>CConvert_ZeneisuToHaneisu.cpp
3>CConvert_ZenkataToHankata.cpp
3>CDecode_Base64Decode.cpp
3>CDecode_UuDecode.cpp
3>convert_util.cpp
3>convert_util2.cpp
3>COpe.cpp
3>COpeBlk.cpp
3>COpeBuf.cpp
3>CProfile.cpp
3>CPropertyManager.cpp
3>CReadManager.cpp
3>CRegexKeyword.cpp
3>CSaveAgent.cpp
3>CSearchAgent.cpp
3>CSelectLang.cpp
3>CSortedTagJumpList.cpp
3>CWriteManager.cpp
3>CRunningTimer.cpp
3>Debug1.cpp
3>Debug2.cpp
3>Debug3.cpp
3>CDialog.cpp
3>CDlgAbout.cpp
3>CDlgCancel.cpp
3>CDlgCompare.cpp
3>CDlgCtrlCode.cpp
3>CDlgDiff.cpp
3>CDlgExec.cpp
3>CDlgFavorite.cpp
3>CDlgFileUpdateQuery.cpp
3>CDlgFind.cpp
3>CDlgGrep.cpp
3>CDlgGrepReplace.cpp
3>CDlgInput1.cpp
3>CDlgJump.cpp
3>CDlgOpenFile.cpp
3>CDlgOpenFile_CommonFileDialog.cpp
3>CDlgOpenFile_CommonItemDialog.cpp
3>CDlgPluginOption.cpp
3>CDlgPrintSetting.cpp
3>CDlgProfileMgr.cpp
3>CDlgProperty.cpp
3>CDlgReplace.cpp
3>CDlgSetCharSet.cpp
3>CDlgTagJumpList.cpp
3>c:\program files (x86)\windows kits\8.1\include\winrt\wrl\wrappers\corewrappers.h(1008): warning C5043: 'Microsoft::WRL::Wrappers::HString::MakeReference': exception specification does not match previous declaration (compiling source file ..\sakura_core\dlg\CDlgOpenFile_CommonItemDialog.cpp)
3>c:\program files (x86)\windows kits\8.1\include\winrt\wrl\wrappers\corewrappers.h(857): note: see declaration of 'Microsoft::WRL::Wrappers::HString::MakeReference' (compiling source file ..\sakura_core\dlg\CDlgOpenFile_CommonItemDialog.cpp)
3>c:\program files (x86)\windows kits\8.1\include\winrt\wrl\wrappers\corewrappers.h(1019): warning C5043: 'Microsoft::WRL::Wrappers::HString::MakeReference': exception specification does not match previous declaration (compiling source file ..\sakura_core\dlg\CDlgOpenFile_CommonItemDialog.cpp)
3>c:\program files (x86)\windows kits\8.1\include\winrt\wrl\wrappers\corewrappers.h(860): note: see declaration of 'Microsoft::WRL::Wrappers::HString::MakeReference' (compiling source file ..\sakura_core\dlg\CDlgOpenFile_CommonItemDialog.cpp)
3>CDlgTagsMake.cpp
3>CDlgWindowList.cpp
3>CDlgWinSize.cpp
3>CBookmarkManager.cpp
3>CDiffManager.cpp
3>CFuncListManager.cpp
3>CModifyManager.cpp
3>CBlockComment.cpp
3>CDocEditor.cpp
3>CDocFile.cpp
3>CDocFileOperation.cpp
3>CDocListener.cpp
3>CDocLocker.cpp
3>CDocOutline.cpp
3>CDocReader.cpp
3>CDocType.cpp
3>CDocTypeSetting.cpp
3>CDocVisitor.cpp
3>CEditDoc.cpp
3>CLineComment.cpp
3>CLayout.cpp
3>CLayoutMgr.cpp
3>CLayoutMgr_DoLayout.cpp
3>CLayoutMgr_New.cpp
3>CLayoutMgr_New2.cpp
3>c:\program files (x86)\windows kits\8.1\include\winrt\wrl\wrappers\corewrappers.h(1008): warning C5043: 'Microsoft::WRL::Wrappers::HString::MakeReference': exception specification does not match previous declaration (compiling source file ..\sakura_core\doc\CEditDoc.cpp)
3>c:\program files (x86)\windows kits\8.1\include\winrt\wrl\wrappers\corewrappers.h(857): note: see declaration of 'Microsoft::WRL::Wrappers::HString::MakeReference' (compiling source file ..\sakura_core\doc\CEditDoc.cpp)
3>c:\program files (x86)\windows kits\8.1\include\winrt\wrl\wrappers\corewrappers.h(1019): warning C5043: 'Microsoft::WRL::Wrappers::HString::MakeReference': exception specification does not match previous declaration (compiling source file ..\sakura_core\doc\CEditDoc.cpp)
3>c:\program files (x86)\windows kits\8.1\include\winrt\wrl\wrappers\corewrappers.h(860): note: see declaration of 'Microsoft::WRL::Wrappers::HString::MakeReference' (compiling source file ..\sakura_core\doc\CEditDoc.cpp)
3>CTsvModeInfo.cpp
3>CDocLine.cpp
3>CDocLineMgr.cpp
3>CAppNodeManager.cpp
3>CDocTypeManager.cpp
3>CFileNameManager.cpp
3>CFormatManager.cpp
3>CHelpManager.cpp
3>CommonSetting.cpp
3>CSakuraEnvironment.cpp
3>CSearchKeywordManager.cpp
3>CShareData.cpp
3>CShareData_IO.cpp
3>CTagJumpManager.cpp
3>DLLSHAREDATA.cpp
3>CBregexp.cpp
3>CBregexpDll2.cpp
3>CDllHandler.cpp
3>CHtmlHelp.cpp
3>CMigemo.cpp
3>CUxTheme.cpp
3>CFuncKeyWnd.cpp
3>CFuncLookup.cpp
3>CKeyBind.cpp
3>Funccode.cpp
3>CBinaryStream.cpp
3>CFile.cpp
3>CFileLoad.cpp
3>CIoBridge.cpp
3>CStream.cpp
3>CTextStream.cpp
3>CZipFile.cpp
3>CCookieManager.cpp
3>CEditorIfObj.cpp
3>CIfObj.cpp
3>CKeyMacroMgr.cpp
3>CMacro.cpp
3>CMacroFactory.cpp
3>CMacroManagerBase.cpp
3>CPPA.cpp
3>CPPAMacroMgr.cpp
3>CSMacroMgr.cpp
3>CWSH.cpp
3>CWSHIfObj.cpp
3>CWSHManager.cpp
3>CMemory.cpp
3>CNative.cpp
3>CNativeA.cpp
3>CNativeW.cpp
3>CRecycledBuffer.cpp
3>CMyWnd.cpp
3>CDlgFileTree.cpp
3>CDlgFuncList.cpp
3>CFuncInfo.cpp
3>CFuncInfoArr.cpp
3>CWordParse.cpp
3>CDllPlugin.cpp
3>CJackManager.cpp
3>CPlugin.cpp
3>CPluginManager.cpp
3>CWSHPlugin.cpp
3>CPrint.cpp
3>CPrintPreview.cpp
3>CPropComBackup.cpp
3>CPropComCustmenu.cpp
3>CPropComEdit.cpp
3>CPropComFile.cpp
3>CPropComFileName.cpp
3>CPropComFormat.cpp
3>CPropComGeneral.cpp
3>CPropComGrep.cpp
3>CPropComHelper.cpp
3>CPropComKeybind.cpp
3>CPropComKeyword.cpp
3>CPropComMacro.cpp
3>CPropComMainMenu.cpp
3>CPropCommon.cpp
3>CPropComPlugin.cpp
3>CPropComStatusbar.cpp
3>CPropComTab.cpp
3>CPropComToolbar.cpp
3>CPropComWin.cpp
3>CMRUFile.cpp
3>CMRUFolder.cpp
3>CMruListener.cpp
3>CRecent.cpp
3>CRecentCmd.cpp
3>CRecentCurDir.cpp
3>CRecentEditNode.cpp
3>CRecentExceptMru.cpp
3>CRecentExcludeFile.cpp
3>CRecentExcludeFolder.cpp
3>CRecentFile.cpp
3>CRecentFolder.cpp
3>CRecentGrepFile.cpp
3>CRecentGrepFolder.cpp
3>CRecentImp.cpp
3>CRecentReplace.cpp
3>CRecentSearch.cpp
3>CRecentTagjumpKeyword.cpp
3>CDlgKeywordSelect.cpp
3>CDlgSameColor.cpp
3>CDlgTypeAscertain.cpp
3>CDlgTypeList.cpp
3>CImpExpManager.cpp
3>CPropTypes.cpp
3>CPropTypesColor.cpp
3>CPropTypesKeyHelp.cpp
3>CPropTypesRegex.cpp
3>CPropTypesScreen.cpp
3>CPropTypesSupport.cpp
3>CPropTypesWindow.cpp
3>CType.cpp
3>CTypeSupport.cpp
3>CType_Asm.cpp
3>CType_Awk.cpp
3>CType_Basis.cpp
3>CType_Cobol.cpp
3>CType_CorbaIdl.cpp
3>CType_Cpp.cpp
3>CType_Dos.cpp
3>CType_Erlang.cpp
3>CType_Html.cpp
3>CType_Ini.cpp
3>CType_Java.cpp
3>CType_Others.cpp
3>CType_Pascal.cpp
3>CType_Perl.cpp
3>CType_Python.cpp
3>CType_Rich.cpp
3>CType_Sql.cpp
3>CType_Tex.cpp
3>CType_Text.cpp
3>CType_Vb.cpp
3>CGraphics.cpp
3>CImageListMgr.cpp
3>CMenuDrawer.cpp
3>CSoundSet.cpp
3>CVisualProgress.cpp
3>CWaitCursor.cpp
3>file.cpp
3>format.cpp
3>input.cpp
3>MessageBoxF.cpp
3>module.cpp
3>ole_convert.cpp
3>os.cpp
3>relation_tool.cpp
3>shell.cpp
3>string_ex.cpp
3>string_ex2.cpp
3>tchar_convert.cpp
3>tchar_printf.cpp
3>tchar_receive.cpp
3>tchar_template.cpp
3>window.cpp
3>CCaret.cpp
3>CEditView.cpp
3>CEditView_Cmdgrep.cpp
3>CEditView_CmdHokan.cpp
3>CEditView_Cmdisrch.cpp
3>CEditView_Command.cpp
3>CEditView_Command_New.cpp
3>CEditView_Diff.cpp
3>CEditView_ExecCmd.cpp
3>CEditView_Ime.cpp
3>CEditView_Mouse.cpp
3>CEditView_Paint.cpp
3>CEditView_Paint_Bracket.cpp
3>CEditView_Scroll.cpp
3>CEditView_Search.cpp
3>CColorStrategy.cpp
3>CColor_Comment.cpp
3>CColor_Found.cpp
3>CColor_Heredoc.cpp
3>CColor_KeywordSet.cpp
3>CColor_Numeric.cpp
3>CColor_Quote.cpp
3>CColor_RegexKeyword.cpp
3>CColor_Url.cpp
3>CRuler.cpp
3>CTextArea.cpp
3>c:\projects\sakura\sakura_core\view\colors\ccolor_numeric.cpp(71): warning C5033: 'register' is no longer a supported storage class
3>c:\projects\sakura\sakura_core\view\colors\ccolor_numeric.cpp(72): warning C5033: 'register' is no longer a supported storage class
3>c:\projects\sakura\sakura_core\view\colors\ccolor_numeric.cpp(73): warning C5033: 'register' is no longer a supported storage class
3>c:\projects\sakura\sakura_core\view\colors\ccolor_numeric.cpp(74): warning C5033: 'register' is no longer a supported storage class
3>c:\projects\sakura\sakura_core\view\colors\ccolor_numeric.cpp(75): warning C5033: 'register' is no longer a supported storage class
3>CTextDrawer.cpp
3>CTextMetrics.cpp
3>CViewCalc.cpp
3>CViewFont.cpp
3>CViewParser.cpp
3>CViewSelect.cpp
3>DispPos.cpp
3>CFigureManager.cpp
3>CFigureStrategy.cpp
3>CFigure_Comma.cpp
3>CFigure_CtrlCode.cpp
3>CFigure_Eol.cpp
3>CFigure_HanSpace.cpp
3>CFigure_Tab.cpp
3>CFigure_ZenSpace.cpp
3>CAutoScrollWnd.cpp
3>CEditWnd.cpp
3>CMainStatusBar.cpp
3>CMainToolBar.cpp
3>CSplitBoxWnd.cpp
3>CSplitterWnd.cpp
3>CTabWnd.cpp
3>CTipWnd.cpp
3>CWnd.cpp
3>CAppMode.cpp
3>CCommandLine.cpp
3>CControlProcess.cpp
3>CControlTray.cpp
3>CNormalProcess.cpp
3>CProcess.cpp
3>CProcessFactory.cpp
3>global.cpp
3>WinMain.cpp
3>CClipboard.cpp
3>CDropTarget.cpp
3>sakura.vcxproj -> C:\projects\sakura\Win32\Debug\sakura.exe
3>=======================
3>postBuild
3>=======================
3>PLATFORM=Win32
3>CONFIGURATION=Debug
3>..\installer\externals\bregonig\bron412.zip -> Win32\Debug\bregonig.dll
3>..\installer\externals\universal-ctags\ctags-2018-09-16_e522743d-x86.zip -> Win32\Debug\ctags.exe
3>Done building project "sakura.vcxproj".
4>------ Build started: Project: tests1, Configuration: Debug Win32 ------
4>Cheking product line version of the Visual Studio.
4>2017
4>find-tools.bat
4>|- CMD_GIT=C:\Program Files\Git\cmd\git.exe
4>|- CMD_7Z=C:\Program Files\7-Zip\7z.exe
4>|- CMD_HHC=C:\Program Files (x86)\HTML Help Workshop\hhc.exe
4>|- CMD_ISCC=C:\Program Files (x86)\Inno Setup 5\ISCC.exe
4>|- CMD_CPPCHECK=C:\Program Files\Cppcheck\cppcheck.exe
4>|- CMD_DOXYGEN=
4>|- CMD_VSWHERE=C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe
4>|- CMD_MSBUILD=C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\amd64\MSBuild.exe
4>|- CMD_CMAKE=C:\Program Files\CMake\bin\cmake.exe
4>end
4>**********************************************************************
4>** Visual Studio 2017 Developer Command Prompt v15.9.15
4>** Copyright (c) 2017 Microsoft Corporation
4>**********************************************************************
4>[vcvarsall.bat] Environment initialized for: 'x86'
4>-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.18362.
4>-- The C compiler identification is MSVC 19.16.27032.1
4>-- The CXX compiler identification is MSVC 19.16.27032.1
4>-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
4>-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
4>-- Detecting C compiler ABI info
4>-- Detecting C compiler ABI info - done
4>-- Detecting C compile features
4>-- Detecting C compile features - done
4>-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
4>-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
4>-- Detecting CXX compiler ABI info
4>-- Detecting CXX compiler ABI info - done
4>-- Detecting CXX compile features
4>-- Detecting CXX compile features - done
4>-- Found PythonInterp: C:/Program Files/Python37/python.exe (found version "3.7.4")
4>-- Looking for pthread.h
4>-- Looking for pthread.h - not found
4>-- Found Threads: TRUE
4>-- Configuring done
4>-- Generating done
4>EXEC : CMake warning :
4>  Manually-specified variables were not used by the project:
4>
4>    CMAKE_BUILD_TYPE
4>
4>
4>-- Build files have been written to: C:/projects/sakura/tests/build/Win32/Debug/googletest
4>.NET Framework 向け Microsoft (R) Build Engine バージョン 15.9.21+g9802d43bc3
4>Copyright (C) Microsoft Corporation.All rights reserved.
4>
4>  Checking Build System
4>  CMake does not need to re-run because C:/projects/sakura/tests/build/Win32/Debug/googletest/CMakeFiles/generate.stamp is up-to-date.
4>  CMake does not need to re-run because C:/projects/sakura/tests/build/Win32/Debug/googletest/googletest/CMakeFiles/generate.stamp is up-to-date.
4>  Building Custom Rule C:/projects/sakura/tests/googletest/googletest/CMakeLists.txt
4>  CMake does not need to re-run because C:/projects/sakura/tests/build/Win32/Debug/googletest/googletest/CMakeFiles/generate.stamp is up-to-date.
4>  gtest-all.cc
4>  gtest.vcxproj -> C:\projects\sakura\tests\build\Win32\Debug\googletest\lib\Debug\gtestd.lib
4>  Building Custom Rule C:/projects/sakura/tests/googletest/googletest/CMakeLists.txt
4>  CMake does not need to re-run because C:/projects/sakura/tests/build/Win32/Debug/googletest/googletest/CMakeFiles/generate.stamp is up-to-date.
4>  gtest_main.cc
4>  gtest_main.vcxproj -> C:\projects\sakura\tests\build\Win32\Debug\googletest\lib\Debug\gtest_maind.lib
4>  Building Custom Rule C:/projects/sakura/tests/googletest/CMakeLists.txt
4>  CMake does not need to re-run because C:/projects/sakura/tests/build/Win32/Debug/googletest/CMakeFiles/generate.stamp is up-to-date.
4>test-cmemory.cpp
4>test-cnative.cpp
4>test-int2dec.cpp
4>test-is_mailaddress.cpp
4>test-mydevmode.cpp
4>test-parameterized.cpp
4>test-sample-disabled.cpp
4>test-sample.cpp
4>test-StdControl.cpp
4>LINK : fatal error LNK1104: cannot open file 'gtest_maind.lib'
4>Done building project "tests1.vcxproj" -- FAILED.
========== Build: 3 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

</details>

Reply:
ちょっと確認してみます。

> %GENERATOR% がダブルクォートで囲まれていますが、それを外して tests1 プロジェクトをビルドすると下記のように出力されました。

`%GENERATOR%` には `Visual Studio 15 2017` のような空白を含む文字列が入るので、ダブルクォートで囲うのは正しいはずです。で、googletestのビルドが通ってるのに、テスト時にリンクが失敗してるように見えますが、再現できるか見てみます。

Reply:
そうそう、リンクが失敗してるんです。

リンカーのオプションの `/LIBPATH:folder` 指定の問題な気がしますが、構成が無駄に複雑になってて自分には突き止める事が出来てません。GoogleTestってマゾ御用達だなって再認識しました。

Reply:
パラメータ展開がうまくいってなかったようです。
冒頭で `GENERATOR=%~2` して辻褄を合わせました。
vs2017(単独)、vs2017(with vs2019)、vs2019の12環境でリビルドして問題なさそうです。

Reply:
要するに `-G "%GENERATOR%"` が意図通りに展開されなくて vs2019 ジェネレータが選択されたっぽいです。

意図した状態） `-G "Visual Studio 15 2017"` 
実際の展開状態） `-G ""Visual Studio 15 2017""` 

きゃーーーｗ

Reply:
最新をPullしてリビルドしたらビルドは通るはずです。
（tests1だけcleanしてビルドでも良いはず。）

Reply:
0ec124781107da403e9d8ac99320de9d5accb251 を pull して git cleanup をして再度確認しました。

Generator で Ninja を使わないようにする為、`tests/googletest.build.cmd` の 24～27行目は rem でコメントアウトをしています。

結局のところリンカエラーで失敗してしまいます。

Reply:
> Generator で Ninja を使わないようにする為、tests/googletest.build.cmd の 24～27行目は rem でコメントアウトをしています。

この条件で vs2019 からビルドしたら `Visual Studio 16 2019` というジェネレータはない、と怒られました。もうちょっと見てみます。

Reply:
何が起こっているか、代替把握できました。

Visual Studio Generatorがビルド成果物を出力するフォルダがズレているからリンクエラーが起きているようです。

Reply:
あと Debug か Release かという指定を行っていないので、自分の環境では以下のような事に…。

![image](https://user-images.githubusercontent.com/1131125/63226554-fe6a6c80-c215-11e9-9dfe-4fdce7c17f4a.png)

Reply:
修正作ってみましたが、vs2019のreleaseビルドでエラーが発生しています。

Reply:
`cmake --help` で Generator の説明を見ると VS2019 だと `-A` オプションでプラットフォーム指定するって書かれてますがそれとは無関係でしょうか？

Reply:
-A は vs generator 固有の仕様で、ジェネレータ名にWin64を含めることで回避できるようです。 2dbd629 で対応しました。

Reply:
何だろう・・・　cl.exe で -Yu と言ったらプリコンパイル済みヘッダーを使う、なんだけどLink.exeにそんなオプションあっただろうか。

<details>
<summary>Output</summary>

```
1>------ Build started: Project: tests1, Configuration: Release Win32 ------
1>find-tools.bat
1>|- CMD_GIT=C:\Program Files\Git\cmd\git.exe
1>|- CMD_7Z=C:\Program Files\7-Zip\7z.exe
1>|- CMD_HHC=C:\Program Files (x86)\HTML Help Workshop\hhc.exe
1>|- CMD_ISCC=
1>|- CMD_CPPCHECK=C:\Program Files\Cppcheck\cppcheck.exe
1>|- CMD_DOXYGEN=
1>|- CMD_VSWHERE=C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe
1>|- CMD_MSBUILD=C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\amd64\MSBuild.exe
1>|- CMD_CMAKE=
1>end
1>**********************************************************************
1>** Visual Studio 2017 Developer Command Prompt v15.9.15
1>** Copyright (c) 2017 Microsoft Corporation
1>**********************************************************************
1>[vcvarsall.bat] Environment initialized for: 'x86'
1>-- Configuring done
1>-- Generating done
1>-- Build files have been written to: C:/work/sakura2019/tests/build/Win32/Release/googletest
1>ninja: no work to do.
1>LINK : fatal error C1007: unrecognized flag '-Yu' in 'p2'
1>LINK : fatal error LNK1257: code generation failed
1>Done building project "tests1.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 4 up-to-date, 0 skipped ==========
```

</details>

Reply:
あ、vs2019で実行してるのにパスがvs2017になっとる・・・。

Reply:
c29b210af1b7fdc1d85f080e338444d9903ade1c を pull して確認しました。

`tests/googletest.build.cmd` の 24～27行目をコメントアウトしていない場合は VS2017 で Win32/x64, Debug/Release 4パターンビルドが通りました。

コメントアウトした場合ですが、Win32, Debug でリビルドすると下記のエラーが出ました。
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB3030	Could not copy the file "D:\projects\sakura\tests\build\Win32\Debug\googletest\\bin\gtestd.pdb" because it was not found.	tests1	C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets	4365	
Error	MSB3030	Could not copy the file "D:\projects\sakura\tests\build\Win32\Debug\googletest\\bin\gtest_maind.pdb" because it was not found.	tests1	C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets	4365	
```

実際に `gtestd.pdb` と `gtest_maind.pdb` が存在するのは自分の環境では Win32, Debug ビルドでは下記の場所です。

```
D:\projects\sakura\tests\build\Win32\Debug\googletest\bin\Debug
```

ところで何で pdb ファイルまでコピーしようとしてるんでしょうか？libファイルの中にデバッグ情報のパスってフルパスで入らなかったですっけ？未確認ですが…。

Reply:
こちらで確認しましたがやはりコピー元のパスが不完全で失敗してしまいますね。 @berryzplus さんの手元ではうまく動いているとしたら不思議だなぁ…。

Reply:
再度試したら、 VS2017 では Win32/x64, Debug/Release の4パターンでビルドが通りました。
VS2019では未確認ですが確かまだ必須要件では無かったような…。。